test/examples/test_kronecker_multitask_sgpr_regression.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+from math import pi
+
+import os
+import random
+import torch
+import unittest
+
+import gpytorch
+from gpytorch.means import ConstantMean, MultitaskMean
+from gpytorch.likelihoods import MultitaskGaussianLikelihood
+from gpytorch.distributions import MultitaskMultivariateNormal
+
+
+# Simple training data: let's try to learn a sine function
+train_x = torch.linspace(0, 1, 100)
+
+# y1 function is sin(2*pi*x) with noise N(0, 0.04)
+train_y1 = torch.sin(train_x * (2 * pi)) + torch.randn(train_x.size()) * 0.1
+# y2 function is cos(2*pi*x) with noise N(0, 0.04)
+train_y2 = torch.cos(train_x * (2 * pi)) + torch.randn(train_x.size()) * 0.1
+
+# Create a train_y which interleaves the two
+train_y = torch.stack([train_y1, train_y2], -1)
+
+
+class MultitaskGPModel(gpytorch.models.ExactGP):
+    def __init__(self, train_x, train_y, likelihood):
+        super(MultitaskGPModel, self).__init__(train_x, train_y, likelihood)
+        self.mean_module = MultitaskMean(ConstantMean(), num_tasks=2)
+        self.covar_module = gpytorch.kernels.MultitaskKernel(
+            gpytorch.kernels.InducingPointKernel(
+                gpytorch.kernels.RBFKernel(), inducing_points=torch.randn(50, 1), likelihood=likelihood
+            ), num_tasks=2, rank=2
+        )
+
+    def forward(self, x):
+        mean_x = self.mean_module(x)
+        covar_x = self.covar_module(x)
+        return MultitaskMultivariateNormal(mean_x, covar_x)
+
+
+class TestKroneckerMultiTaskSGPRRegression(unittest.TestCase):
+    def setUp(self):
+        if os.getenv("UNLOCK_SEED") is None or os.getenv("UNLOCK_SEED").lower() == "false":
+            self.rng_state = torch.get_rng_state()
+            torch.manual_seed(0)
+            if torch.cuda.is_available():
+                torch.cuda.manual_seed_all(0)
+            random.seed(0)
+
+    def tearDown(self):
+        if hasattr(self, "rng_state"):
+            torch.set_rng_state(self.rng_state)
```

Comment:
Orthogonal to this PR: It seems like this could be generally useful and we may want to factor it out into sth like a `GPyTorchTestCase`?

Reply:
I think this functionality is already in [BaseTestCase](https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/test/base_test_case.py), so we probably should be using it 😅

Reply:
I substituted "inducing_point_kernel_added_loss_term.py" in "gpytorch/mlls" with this revised file here and run the code snippet that I submitted in GitHub (#2113). But it didn't work. Did I do something wrong or is the issue not solved yet?

Reply:
@GHU2021 this PR is being merged in today. If you are still experiencing issues, please open up another issue, rather than commenting on this PR. Please also provide more information than "But it didn't work."